src/components/ReportInfoView.js

```diff
@@ -79,13 +83,19 @@ class ReportInfoView extends React.Component {
     }
 
     this.setState({
-      didClickNav: true,
       sectionRequiredErrors: requiredErrors
     });
+  }
 
-    if (requiredErrors.length < 1 && this.state.sectionFormatErrors.length < 1) {
-      this.props.handlePageChange(path);
-    }
+  handlePageChange = (path) => {
+    this.setDidClickNav();
```

Comment:
Does order of events matter here? `setDidClickNav` calls `setState`, which is asynchronous, so you don't have a guarantee of `didClickNav` being updated before the next step in this function.